File: c++/best-time-to-buy-and-sell-stock.cc
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0;
        int buy = prices[0];

        for (int price : prices) {
            if (price < buy) {
                buy = price;
            } else {
                maxProfit = max(maxProfit, price - buy);
            }
        }
        return maxProfit;
    }
};

File: python/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2:
            return 0

        maxProfit = 0
        left = 0
        right = 1

        while left < right and right < len(prices):
            buy = prices[left]
            sell = prices[right]

            if buy >= sell:
                left += 1
                if left == right:
                    right += 1
            else:
                maxProfit = max(maxProfit, sell - buy)
                right += 1

        return maxProfit
